### ISSUE TITLE
Installation issue: py-tensorflow@2.4.1

### ISSUE THREAD

**Issue by nkgh77** [LLNL/spack#28903]
### Steps to reproduce the issue

spack install py-tensorflow~cuda+mkl

### Information on your system

* **Spack:** 0.17.1-1163-8c943261a2
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-skylake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/8054094/spack-build-env.txt)
[spack-build-01-configure-out.txt](https://github.com/spack/spack/files/8054100/spack-build-01-configure-out.txt)
[spack-build-out.txt.gz](https://github.com/spack/spack/files/8054104/spack-build-out.txt.gz)

@adamjstewart 
@aweits

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by adamjstewart** [comment]
Possibly a duplicate of #28612. Can you try building with an older version of protobuf and see if that works?
```console
$ spack install py-tensorflow~cuda+mkl ^protobuf@:3.14
```
@aweits @glennpj maybe we should fix this in a separate PR since #27138 is taking a while.

**Comment by glennpj** [comment]
I believe that #27138 is only waiting on #28621.

**Comment by nkgh77** [comment]
Thank you. This worked!
> ```
> $ spack install py-tensorflow~cuda+mkl ^protobuf@:3.14
> ```
